services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -139,10 +139,20 @@ CloseableHttpAsyncClient getHttpAsyncClient() {
     return httpAsyncClient;
   }
 
-  protected static boolean whetherStoreReadyToBeCleanup(Store store, long defaultBackupVersionRetentionMs, Time time) {
+  protected static boolean whetherStoreReadyToBeCleanup(
+      Store store,
+      long defaultBackupVersionRetentionMs,
+      Time time,
+      int currentVersion) {
     if (store.getCurrentVersion() == NON_EXISTING_VERSION || store.getVersions().size() < 2) {
       return false;
     }
+
+    if (store.getVersion(currentVersion).get().getRepushSourceVersion() > NON_EXISTING_VERSION
+        && store.getVersions().size() > 2) {
```

Comment:
So shall we delete the source version even the total number of versions are `2`?

Reply:
If we add a delay before deleting the repush source version, I guess we can delete if total  number of versions are 2.